source/utils/types.ts

```diff
@@ -0,0 +1,21 @@
+import {IncomingMessage} from 'http';
+import {RequestOptions} from 'https';
+import {PCancelable} from 'p-cancelable';
+import {Hooks} from '../known-hook-events';
+
+export type Method = 'GET' | 'PUT' | 'HEAD' | 'DELETE' | 'OPTIONS' | 'TRACE' | 'get' | 'put' | 'head' | 'delete' | 'options' | 'trace';
+
+export interface Options extends RequestOptions {
+	hooks?: Partial<Hooks>;
+	decompress?: boolean;
+	encoding?: BufferEncoding | null;
+	method?: Method;
+	[ key: string ]: unknown | Options;
```

Comment:
Is the `Options` type needed here?

Reply:
We should also add a comment that the line should be removed in the future when the TS migration is complete.